test/ERC4626.t.sol

```diff
@@ -80,3 +80,86 @@ contract SparkVaultERC4626Test is ERC4626Test, SparkVaultTestBase {
     }
 
 }
+
+contract SparkVaultERC4626Test is SparkVaultERC4626StandardTest {
+
+    address user1 = makeAddr("user1");
+    address user2 = makeAddr("user2");
+
+    event Referral(uint16 indexed referral, address indexed owner, uint256 assets, uint256 shares);
+
+    // Do some deposits to get some non-zero state
+    function setUp() public override {
+        super.setUp();
+
+        vm.prank(admin);
+        vault.setSsrBounds(ONE_PCT_SSR, FOUR_PCT_SSR);
+
+        vm.prank(setter);
+        vault.setSsr(FOUR_PCT_SSR);
+
+        deal(address(asset), user1, 1_000_000e6);
+
+        vm.startPrank(user1);
+        asset.approve(address(vault), 1_000_000e6);
+        vault.deposit(1_000_000e6, user1);
+        vm.stopPrank();
+
+        skip(1 days);
+    }
+
+    function test_depositWithReferral() public {
+        uint16 referral = 1;
+
+        uint256 assets = 1_000_000e18;
+        uint256 shares = vault.previewDeposit(assets);
+
+        deal(address(asset), user2, assets);
+
+        assertEq(vault.balanceOf(user2), 0);
+        assertEq(vault.totalSupply(),    986_721.777104e6);
+        assertEq(asset.balanceOf(user2), assets);
```

Comment:
asset before, add assertion for asset balance of vault as well

Reply:
added